src/main/java/com/feedzai/commons/sql/abstraction/engine/impl/H2Engine.java

```diff
@@ -385,10 +385,13 @@ protected MappedEntity createPreparedStatementForInserts(final DbEntity entity)
         final List<String> values = new ArrayList<>();
         final List<String> columnsWithAutoInc = new ArrayList<>();
         final List<String> valuesWithAutoInc = new ArrayList<>();
+        String columnWithAutoIncName = null;
         for (final DbColumn column : entity.getColumns()) {
             columnsWithAutoInc.add(quotize(column.getName()));
             valuesWithAutoInc.add("?");
-            if (!column.isAutoInc()) {
+            if (column.isAutoInc()) {
```

Comment:
What if we have more than 1 auto_increment column? Shouldn't we save these names in a list and in the #doPersist use for example the first name that appears on the list in position 0? We need to specify in the documentation that we assume that the first column with auto_increment is the one that contains the entity ID

Reply:
If we have more than one column as auto_increment we will end up assuming that the one that provides entity id is the last one. We can also continue using a string instead of a list but don't set this string more than once.

Reply:
I think we should keep using just a simple `String` because if we added the functionality to retrieve every auto incremented value,  it would break the current API, right?
So I think the best approach is either return the auto incremented value from the first or last column and state it in the Javadoc.

Reply:
I agree with having just a string since we just want to retrieve one auto incremented column as the entity id. I think it makes sense to be the first auto incremented column that we observe. Normally primary keys are defined first in CREATE TABLE statements